### PR TITLE
Run activation behavior hooks from the event dispatch algorithm 

### LINKED ISSUE #2449: clicking span inside a submit button does not submit the form
### Basic info:

- **Node.js version:** >=8
- **jsdom version:** 13

### Minimal reproduction case

[Repl.it](https://repl.it/@kentcdodds/jsdom-submit-bug)

```js
const {JSDOM} = require('jsdom')


const {window, window: {MouseEvent, document}} = new JSDOM(`
<!doctype html>
<html>
  <body>
    <form>
      <button type="submit">
        <span>submit</span>
      </button>
    </form>
  </body>
</html>
`)

const form = document.querySelector('form')
const submit = document.querySelector('span')

let calls = 0

form.addEventListener('submit', event => {
  event.preventDefault()
  console.log('submitted')
  calls++
})

const clickEvent = new MouseEvent('click', {bubbles: true, cancelable: true, button: 0})
submit.dispatchEvent(clickEvent)

console.assert(calls === 1, 'the form was not submitted')
```

### How does similar code behave in browsers?

[jsbin](https://jsbin.com/recekor/edit?html,js,console)

In the browser, firing a click event on the span within a submit button will call the form's submit handler. In jsdom it does not.

Note that if I select the button instead of the span and fire the click event on the button then the form is submitted just fine. It's just when I select something within the button.

Real world situation: https://github.com/kentcdodds/react-testing-library/issues/234

Thanks!

### PR DESCRIPTION
**Pull request by Zirro**
This adds steps regarding activation behavior from the [Dispatch an event](https://dom.spec.whatwg.org/#concept-event-dispatch) algorithm, replacing older code where the activation didn't bubble correctly.
I've also renamed the existing hooks to match the names used in the specification.

This fixes #2449 and #2367.

### CODE REVIEW

File: lib/jsdom/living/nodes/HTMLElement-impl.js [hunk 1]
```diff
@@ -18,25 +18,8 @@ class HTMLElementImpl extends ElementImpl {
     this._clickInProgress = false;
   }
 
-  // Add default event behavior (click link to navigate, click button to submit
-  // form, etc). We start by wrapping dispatchEvent so we can forward events to
-  // the element's default functions (only events that did not incur
-  // preventDefault).
   dispatchEvent(event) {
```
Comment (TimothyGu) [inline lib/jsdom/living/nodes/HTMLElement-impl.js]: Can we remove this method now that it purely forwards?

Comment (Zirro) [reply lib/jsdom/living/nodes/HTMLElement-impl.js]: Yes, thanks for the suggestion.

**Comment by Zirro** [comment]
Updated to add `_hasActivationBehavior` instead of directly checking for the existence of the method, as it occurred to me that it'll need to be conditional for `<summary>` which uses `HTMLElement`.

**Comment by domenic** [comment]
Nice work. Let's wait to merge this until a WPT roll though, just out of principle :).

**Pull request merged**